### PR TITLE
destination-specification: add supportsNormalization and supportsDBT attributes

### PR DESCRIPTION
**Pull request by subodh1810**
Part of https://github.com/airbytehq/airbyte/pull/3624

### CODE REVIEW

File: airbyte-integrations/connectors/destination-e2e-test/src/main/resources/spec.json [hunk 1]
```diff
@@ -1,6 +1,8 @@
 {
   "documentationUrl": "https://example.com",
   "supportsIncremental": true,
+  "supportsNormalization": true,
```
Comment (subodh1810) [inline airbyte-integrations/connectors/destination-e2e-test/src/main/resources/spec.json:4]: I was not sure about `destination-e2e-test` spec

File: airbyte-integrations/connectors/destination-jdbc/src/main/resources/spec.json [hunk 2]
```diff
@@ -1,6 +1,8 @@
 {
   "documentationUrl": "https://docs.airbyte.io/integrations/destinations/postgres",
   "supportsIncremental": true,
+  "supportsNormalization": true,
```
Comment (subodh1810) [inline airbyte-integrations/connectors/destination-jdbc/src/main/resources/spec.json:4]: Not sure about `destination-jdbc` it seems like it refers to postgres

File: airbyte-integrations/connectors/destination-jdbc/src/main/resources/spec.json [hunk 3]
```diff
@@ -1,6 +1,8 @@
...
+  "supportsDBT": true,
```
Comment (ChristopheDuong) [inline airbyte-integrations/connectors/destination-jdbc/src/main/resources/spec.json:5]: I would set both of these to false for the generic jdbc destination

Then, it is overridden in each destination such as postgres etc anyway

File: airbyte-protocol/models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml [hunk 4]
```diff
@@ -236,6 +236,12 @@ definitions:
         description: If the connector supports incremental mode or not.
         type: boolean
       # Destination Connectors Properties
+      supportsNormalization:
+        description: If the connector supports normalization or not.
```
Comment (ChristopheDuong) [inline airbyte-protocol/models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml:240]: Can we add a comment:

    Normalization is currently implemented using dbt so it requires `supportsDBT` to be true for this to be true.

**Review by ChristopheDuong — APPROVED** [review]

File: airbyte-integrations/connectors/destination-e2e-test/src/main/resources/spec.json (hunk 1 above)
Comment (ChristopheDuong) [reply airbyte-integrations/connectors/destination-e2e-test/src/main/resources/spec.json:4]: let's default to false for both

**Comment by michel-tricot** [comment]
What behavior difference do I expect between dbt and normalization?

**Comment by subodh1810** [comment]
@michel-tricot 
supportsDBT represents whether the destination has support for DBT or not, if true then it would allow users to write custom transformations
supportsNormalization represents whether Airbyte's basic normalization works for the destination or not. 

@ChristopheDuong correct me if am wrong or if I missed anything.

**Comment by ChristopheDuong** [comment]
> @michel-tricot
> supportsDBT represents whether the destination has support for DBT or not, if true then it would allow users to write custom transformations
> supportsNormalization represents whether Airbyte's basic normalization works for the destination or not.
> 
> @ChristopheDuong correct me if am wrong or if I missed anything.

Yes for example today, destinations such as MySql or MSSQL do support dbt (through community adapters) so custom dbt transformations for them can be written/used
- but since we haven't implemented normalization for them yet, they would have supportNormalization = false

**Pull request merged**